src/jit/inline.cpp

```diff
@@ -697,7 +697,9 @@ void InlineResult::Report()
 
         InlineObservation obs = m_Policy->GetObservation();
 
-        if ((m_Callee != nullptr) && (obs != InlineObservation::CALLEE_IS_NOINLINE))
+        if ((m_Callee != nullptr) && 
+            (obs != InlineObservation::CALLEE_IS_NOINLINE) && 
+            (obs != InlineObservation::CALLEE_DOES_NOT_RETURN))
```

Comment:
I presume you're not propagating here so that the jit rereads the IL for each call? It would be good noet that here and link to the issue you opened for recording noinline deductions with the runtime.

As with my note above, we'd really like to ensure that the inlinee is small here to avoid repeatedly scanning a large inlinee that must throw.